File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-36.js
class Solution {
    /**
     * @param {string} s
     * @param {number} k
     * @return {number}
     */
    characterReplacement(s, k) { 
        const c = {} 

        let l = 0
        let maxCount = 0 
        let res = 0
        for (let r = 0; r < s.length; r ++) {
            let cur = s[r] 

            c[cur] = ( c[cur] || 0 ) + 1 
            maxCount = Math.max(maxCount, c[cur]) 

            while(( ( r - l + 1) - maxCount) > k ) {
                let deleted = s[l] 
                c[deleted] -- 
                l ++
            } 

            res = Math.max(res, ( r - l + 1 ))
        }

        return res
    }
}

File: Data Structures & Algorithms/house-robber-ii/submission-36.py
class Solution:
    def rob(self, nums: List[int]) -> int:  
        return max(nums[0], self.helper(nums[1: ]), self.helper(nums[:-1]))
        

    def helper(self, nums): 
        r1 = 0 
        r2 = 0 
        for num in nums: 
            cur = max(num + r1, r2) 
            r1 = r2 
            r2 = cur 
        return r2
        